homeassistant/components/velbus/select.py

```diff
@@ -0,0 +1,46 @@
+"""Support for Velbus select."""
+from velbusaio.channels import SelectedProgram
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import VelbusEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Velbus select based on config_entry."""
+    await hass.data[DOMAIN][entry.entry_id]["tsk"]
+    cntrl = hass.data[DOMAIN][entry.entry_id]["cntrl"]
+    entities = []
+    for channel in cntrl.get_all("select"):
+        entities.append(VelbusSelect(channel))
+    async_add_entities(entities)
+
+
+class VelbusSelect(VelbusEntity, SelectEntity):
+    """Representation of a select option for velbus."""
+
+    _channel: SelectedProgram
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(
+        self,
+        channel: SelectedProgram,
+    ) -> None:
+        """Initialize a select Velbus entity."""
+        super().__init__(channel)
+        self._attr_options = self._channel.get_options()
+        self._attr_current_option = self._channel.get_selected_program()
```

Comment:
I would suggest you use the property. This implementation would make `current_option` static

Reply:
If options could change then I suggest same there

Reply:
the options can't change, this is linked to the modules, so there is no way in changing these names.

Reply:
Is the options known so they possibly should be translated?

Reply:
I would not do a translation, in velbus (on the modules) these names will never be translated, so these names are static there.
if we translate them in hass, whats shown in the modules and what is show in hass will not be the same